tedana/selection/select_comps.py

```diff
@@ -71,828 +46,298 @@ def selcomps(seldict, comptable, mmix, mask, ref_img, manacc, n_echos, t2s, s0,
-                           mmix_kurt_z_max < 2.5])
-        new_acc = np.intersect1d(orphan, all_comps[temp == 6])
-        comptable.loc[new_acc, 'rationale'] += 'Saved at the last second;'
-        acc_comps = np.union1d(acc_comps, new_acc)
-        # ign = np.setdiff1d(all_comps, list(acc_comps)+list(midk)+list(rej))
-        orphan = np.setdiff1d(all_comps, (list(acc_comps) + list(ign) +
-                                          list(midk) + list(rej)))
-    comptable.loc[orphan, 'classification'] = 'ignored'
-    comptable.loc[orphan, 'rationale'] += 'orphan;'
-
-    if savecsdiag:
-        diagstep_keys = ['Rejected components', 'Kappa-Rho cut point', 'Kappa cut',
-                         'Rho cut', 'DBSCAN failed to converge', 'Kappa-Rho guess',
-                         'Dice rejected', 'rej_supp', 'to_clf',
-                         'Mid-kappa components', 'svm_acc_fail', 'toacc_hi', 'toacc_lo',
-                         'Field artifacts', 'Physiological artifacts',
-                         'Miscellaneous artifacts', 'acc_comps', 'Ignored components']
-        diagstep_vals = [list(rej), KRcut.item(), Kcut.item(), Rcut.item(),
-                         dbscanfailed, list(KRguess), dice_rej,
-                         list(rej_supp), list(to_clf), list(midk),
-                         svm_acc_fail, list(toacc_hi), list(toacc_lo),
-                         list(field_art), list(phys_art),
-                         list(misc_art), list(acc_comps), list(ign)]
-
-        with open('csstepdata.json', 'w') as ofh:
-            json.dump(dict(zip(diagstep_keys, diagstep_vals)), ofh,
-                      indent=4, sort_keys=True, default=str)
-        allfz = np.array([Tz, Vz, Ktz, KRr, cnz, Rz, mmix_kurt, fdist_z])
-        np.savetxt('csdata.txt', allfz)
+    comptable.loc[ign, 'rationale'] += 'low variance;'
+    acc = np.setdiff1d(acc, ign)
+
+    """
+    Step 5: Scrub the set
+    """
+    if len(acc) > len(good_guess):
+        # Recompute the midk steps on the limited set to clean up the tail
+        d_table_rank = np.vstack([
+            len(acc) - stats.rankdata(comptable.loc[acc, 'kappa'], method='ordinal'),
+            len(acc) - stats.rankdata(comptable.loc[acc, 'dice_FR2'], method='ordinal'),
+            len(acc) - stats.rankdata(comptable.loc[acc, 'tt0'], method='ordinal'),
+            stats.rankdata(countnoise[acc], method='ordinal'),
+            len(acc) - stats.rankdata(comptable.loc[acc, 'countsigFR2'], method='ordinal')]).T
+        comptable['d_table_score_scrub'] = np.nan
+        comptable.loc[acc, 'd_table_score_scrub'] = d_table_rank.sum(1)
+        num_acc_guess = int(np.mean([
+            np.sum((comptable.loc[acc, 'kappa'] > kappa_elbow) &
+                   (comptable.loc[acc, 'rho'] < rho_elbow)),
+            np.sum(comptable.loc[acc, 'kappa'] > kappa_elbow)]))
+        conservative_guess = num_acc_guess * n_decision_metrics / RESTRICT_FACTOR
+
+        # candartA
+        candartA = np.intersect1d(
```

Comment:
`candartA` ? Any idea what that stands for ?

Reply:
Rejection candidate based on artifact type A, is my guess.